devtools/maven/src/main/java/io/quarkus/maven/GenerateCodeMojo.java

```diff
@@ -50,6 +52,12 @@ protected boolean beforeExecute() throws MojoExecutionException, MojoFailureExce
     @Override
     protected void doExecute() throws MojoExecutionException, MojoFailureException {
         String projectDir = mavenProject().getBasedir().getAbsolutePath();
+        Path targetClasses = mavenProject().getBasedir().toPath().resolve("target/classes");
```

Comment:
Ideally, it should be `mavenProject().getBuild().getOutputDirectory()` to make specific assumptions about the classes dir.
What this means though is the execution flow is not expected. This directory is supposed to be created in `QuarkusBootstrapProvider`. What was the actual issue and stacktrace?

Reply:
The stack trace for quarkus-maven-plugin:2.5.0.Final is in https://github.com/quarkusio/quarkus/issues/21718
I think the problem is that in `QuarkusBootstrapProvider.QuarkusAppBootstrapProvider.bootstrapApplication(QuarkusBootstrapMojo, LaunchMode)` `doBootstrap(mojo, mode)` is not called with a warmed up daemon. See here https://github.com/quarkusio/quarkus/blob/2.5.0.Final/devtools/maven/src/main/java/io/quarkus/maven/QuarkusBootstrapProvider.java#L216

Reply:
Isn't it the actual problem?

Reply:
The root cause is actually, that the BootstrapSessionListener is not called unless the plugin is declared with `<extensions>true</extensions>`. Otherwise the `appBootstrapProviders` cache is not cleared upon Maven build termination and thus with the next invocation mvnd re-uses the old cache, which in turn prevents QuarkusBootstrapProvider to create `target/classes`. 
Actually, I could live with having `<extensions>true</extensions>` in Camel Quarkus, but the question is whether the above behavior is expected?

In ce23eba I tried to use EventSpy and register it manually, so that end users do not need to use `<extensions>true</extensions>`, but EventSpy is not visible from the mojo class loader for some reason.

Reply:
If extensions aren't enabled then the assumption was the process would terminate along with the VM, which isn't the case with the mvnd. I am wondering whether we can detect the end of the build w/o enabling extensions to clear the cache.

Reply:
> I am wondering whether we can detect the end of the build w/o enabling extensions to clear the cache.

That was my intention in ce23eba. Strange that it works in mvnd, but not in plain mvn. It is really strange that maven-core is not visible from the mojo class loader.